src/Hl7.Fhir.Support.Poco/Rest/SearchParams.cs

```
@@ -0,0 +1,418 @@
+            var result = new UriParamList();
+
+            if (!String.IsNullOrEmpty(Query)) result.Add(Tuple.Create(SEARCH_PARAM_QUERY, Query));
+            if (!String.IsNullOrEmpty(Text)) result.Add(Tuple.Create(SEARCH_PARAM_TEXT, Text));
+            if (!String.IsNullOrEmpty(Content)) result.Add(Tuple.Create(SEARCH_PARAM_CONTENT, Content));
+            if (Count != null) result.Add(Tuple.Create(SEARCH_PARAM_COUNT, Count.Value.ToString()));
+            if (Include.Any()) result.AddRange(Include.Select(i => Tuple.Create(SEARCH_PARAM_INCLUDE, i)));
+            if (RevInclude.Any()) result.AddRange(RevInclude.Select(i => Tuple.Create(SEARCH_PARAM_REVINCLUDE, i)));
+            if (Sort.Any()) result.Add(createSortParam(Sort));
+            if (Summary != null) result.Add(Tuple.Create(SEARCH_PARAM_SUMMARY, Summary.Value.ToString().ToLower()));
+            if (!String.IsNullOrEmpty(Filter)) result.Add(Tuple.Create(SEARCH_PARAM_FILTER, Filter));
+            if (Contained != null) result.Add(Tuple.Create(SEARCH_PARAM_CONTAINED, Contained.Value.ToString().ToLower()));
+            if (ContainedType != null) result.Add(Tuple.Create(SEARCH_PARAM_CONTAINEDTYPE, ContainedType.Value.ToString().ToLower()));
+            if (Elements.Any()) result.Add(Tuple.Create(SEARCH_PARAM_ELEMENTS, String.Join(",",Elements)));
+
+            result.AddRange(Parameters);
+            return result;
+
+            Tuple<string,string> createSortParam(IList<Tuple<string,SortOrder>> sorts)
+            {
+                var values = 
+                    from s in sorts
+                    let orderPrefix = s.Item2 == SortOrder.Descending ? "-" : ""
+                    select orderPrefix + s.Item1;
+                return new Tuple<string,string>(SEARCH_PARAM_SORT, String.Join(",", values));
+            }
+        }
+        
+        [Obsolete("Use the Parameters.ToSearchParameters() method instead.", true)]
+        public static SearchParams FromParameters(object parameters)
```

Obsolete methods throwing is not my favorite, though I think I asked George to do it this way. What do you guys think?

---

Hmmm doesn't deserve to win the beauty pageant.
But we can't implement the older code, since it requires the Poco. Shall we just leave it this way? It throws an error anyway instead of a warning.